solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -510,14 +510,18 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
   }
 
   private void tagRequestWithRequestId(ResponseBuilder rb) {
-    String rid = getRequestId(rb.req);
-    if (StringUtils.isBlank(rb.req.getParams().get(CommonParams.REQUEST_ID))) {
-      ModifiableSolrParams params = new ModifiableSolrParams(rb.req.getParams());
-      params.add(CommonParams.REQUEST_ID, rid);//add rid to the request so that shards see it
-      rb.req.setParams(params);
-    }
-    if (rb.isDistrib) {
-      rb.rsp.addToLog(CommonParams.REQUEST_ID, rid); //to see it in the logs of the landing core
+    final String disableFlag = rb.req.getParams().get(CommonParams.DISABLE_REQUEST_ID);
```

Comment:
I don't think this flag should be a request parameter; I think it should be a "Cluster Property".

Secondly, if you do need to get this from a SolrParams, see `getBool` method, accepting a default.

Reply:
Man you're quick on the reviews.

I like a SolrParam better than a cluster property.  Users might well have reasons to enable/disable it differently based on the source or type of query.  A Cluster Property doesn't offer that sort of granularity (at least, as I understand them).

Reply:
Having to disable one parameter showing up by adding another parameter kinda defeats the point.  To some people, this new parameter is needless noise.  To them, your solution is to tell them to add more noise (another param) to make that one go away.  I don't understand why users would toggle this on a query-by-query basis.  I suspect people holistically like this new ID on all their requests, or they don't.  But maybe you have something in mind?  I suppose it *could* be both -- you set the default as a cluster property but allow an override for the request.  Hmm; yeah I rather like that.

Reply:
> I like a SolrParam better than a cluster property. Users might well have reasons to enable/disable it differently based on the source or type of query.

If users want full per-query control, they don't need a Solr feature at all, they can just start including `whatever=1234` in their requests and track it in their logs. :)